cranelift/interpreter/src/value.rs

```diff
@@ -223,8 +230,16 @@ impl Value for DataValue {
         }
     }
 
-    fn vector(_v: [u8; 16], _ty: Type) -> ValueResult<Self> {
-        unimplemented!()
+    fn vector(v: [u8; 16], ty: Type) -> ValueResult<Self> {
+        assert!(ty.is_vector());
```

Comment:
This might need to be more strict:
```suggestion
        assert!(ty.is_vector() && elem_ty.bytes() == 16);
```